src/Nethermind/Nethermind.Runner/Program.cs

```diff
@@ -249,6 +255,30 @@ private static void Run(string[] args)
         }
     }
 
+    private static string CheckForDuplicateArgs(string[] args)
+    {
+        Dictionary<string, string> argPairs = new Dictionary<string, string>();
+        foreach (var arg in args){
+            var key = "";
+            if (arg.StartsWith("--"))
+            {
+                key = arg.Substring(2);
+                if(argPairs.ContainsKey(key)){
+                    return key;
+                }
+                argPairs[key] = ""; 
+            }
+            else if (arg.StartsWith("-"))
+            {
+                key = arg.Substring(1);
+                if(argPairs.ContainsKey(key)){
+                    return key;
+                }
+                argPairs[key] = ""; 
+            }
+        }
+        return string.Empty;
+    }
```

Comment:
```suggestion
    private static IEnumerable<ReadOnlyMemory<char>> GetDuplicateArguments(string[] args)
    {
        static ReadOnlyMemory<char> GetArgumentName(string arg) => arg.StartsWith("--") ? arg.AsMemory(2) : arg.StartsWith('-') ? arg.AsMemory(1) : ReadOnlyMemory<char>.Empty;
        return args.GroupBy(GetArgumentName).Where(g => g.Count() > 1).Select(g => g.Key);
    }

```

Reply:
Looks so cool and compact now. Learnt a few new things with these changes.